src/main/java/com/salam/task_manager/services/TaskService.java

```diff
@@ -0,0 +1,35 @@
+package com.salam.task_manager.services;
+
+import com.salam.task_manager.dto.TaskDto;
+import com.salam.task_manager.exception.ResourceNotFoundException;
+import com.salam.task_manager.mapper.TaskMapper;
+import com.salam.task_manager.models.TaskModel;
+import com.salam.task_manager.models.user.User;
+import com.salam.task_manager.repository.TaskRepository;
+import com.salam.task_manager.repository.UserRepository;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@AllArgsConstructor
+public class TaskService {
+
+    private final TaskRepository taskRepository;
+    private final UserRepository userRepository;
+    private final TaskMapper taskMapper;
+
+    @Transactional
```

Comment:
Why Transactional here?

Reply:
To ensure that any database operations performed within the method will be executed within a transaction. If the transaction is successful, the changes will be committed to the database. If an error occurs and the transaction is rolled back, the changes will not be persisted in the database.
Source: [https://nikhilsukhani.medium.com/transactional-annotation-in-spring-boot-ae76307fcd26](url)